spec/rubocop/cop/lint/redundant_safe_navigation_spec.rb

```diff
@@ -125,4 +125,89 @@
       do_something if foo&.to_d
     RUBY
   end
+
+  it 'registers an offense and corrects when `.&` is used in `.to_h` coversion with default' do
+    expect_offense(<<~RUBY)
+      foo&.to_h || {}
+         ^^^^^^^^^^^^ Redundant safe navigation detected.
```

Comment:
In such cases, adjusting the message a bit would likely prevent user confusion. e.g.:

```suggestion
         ^^^^^^^^^^^^ Redundant safe navigation with default value detected.
```